src/runtime/object.cpp

```diff
@@ -1611,9 +1611,23 @@ extern "C" LEAN_EXPORT object * lean_mk_string_from_bytes(char const * s, size_t
 }
 
 extern "C" LEAN_EXPORT object * lean_mk_string(char const * s) {
+    size_t len = strlen(s);
+    if (validate_utf8((const uint8_t *)s, len)) {
```

Comment:
Why is this a separate step, can't it be combined with `utf8_strlen`? Do we even need the mk_string/mk_utf8_string distinction in that case?

Reply:
We can do `utf8_strlen` at the same time as validation, but I think there are still use cases for "I know this is UTF8", in particular when doing the check beforehand so as to give a custom error message. Also, from the looks of it `utf8_strlen` will be optimized to a faster function than `validate_utf8`, so there is likely some performance advantage to using only `utf8_strlen` when there are reasons to believe the input is already utf8. This happens in `String.set` for example (although actually in that example we can do even better since we know what the length is already).